bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/ThingTypeRegistry.java

```diff
@@ -151,7 +151,11 @@ public ChannelType getChannelType(ChannelUID channelUID) {
      */
     public ChannelType getChannelType(ChannelUID channelUID, Locale locale) {
         ThingType thingType = this.getThingType(channelUID.getThingTypeUID(), locale);
-        return thingType != null ? thingType.getChannelType(channelUID) : null;
+        if (thingType != null) {
+            ChannelTypeUID channelTypeUID = thingType.getChannelTypeUID(channelUID);
+            return TypeResolver.resolve(channelTypeUID);
```

Comment:
Well, as the whole thing type meta model is optional, it is possible. You can define a channel in the Thing DSL without an according channel type. So I would not log a warning or any other message here. If the caller expects the channel type to be there, it can log it (as the ThingFactory) does for example. Same applies for the previous comment.